src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -0,0 +1,441 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.Packaging;
+using NuGet.ProjectManagement;
+using NuGet.ProjectModel;
+using NuGet.RuntimeModel;
+using NuGet.Shared;
+using NuGet.Versioning;
+using NuGet.VisualStudio;
+using static NuGet.Frameworks.FrameworkConstants;
+
+namespace NuGet.SolutionRestoreManager
+{
+    internal class VSNominationUtilities
```

Comment:
while we're refactoring this part, you may want to extend this from an interface as well which will help with better unit/ func testing of this MEF class

Reply:
I'm not sure what you mean by that. 

It's a class with static methods, but there are lot of high level tests VsSolutionRestoreServiceTests.cs. 

Did you mean add unit tests for the specific methods in this class? 
I made sure that the tests in the above class cover every code path.